skrf/network.py

```diff
@@ -336,6 +336,11 @@ def _generated_functions(cls) -> dict[str, tuple[Callable, str, str]]:
     Default interpolation method.
     """
 
+    noise_fill_value = np.nan
```

Comment:
probably should be defined in `constants.py`.@FranzForstmayr , what do you think?

Reply:
Probably yes, i need to check as Literals wants to be more or less static without calculation

Reply:
Hmm, I don't see the point why defining this on class level, as changing this would have global effect. Any thoughts against setting the `fill_level` to `np.nan` directly at lines 1471 and 1479?

Reply:
We could hard code it, and I'd be happy with the values, but I could see some users wanting to change how the filling is done and what the array is filled with. Why was `Network.noise_interp_kind` originally added as a class variable if it was always going to be fixed? Shouldn't it have just been hard coded in the few places it's used?

I'm not sure I understand the concern with allowing instance-level modifications to these two values. I feel like 99% of people will never change it, but the interpolation method can be used in so many ways, it seems silly to lock it down so much. I'm not a software engineer, and I'm unclear on what the Literals issue is, that's beyond my programming experience.

Reply:
This has nothing to do with the literals, I assumed this wrongly because of the linked PR. :)

You are right, `noise_interp_kind` should be a instance variable as well IMO. What about an optional kwarg? Like encoding is handled currently? Probably for both variables in question?

Reply:
Totally, I think that would be a great way to handle it, but `Network.n` is a property, not a method, so wouldn't we have to change the way it's accessed? That seems like a rather large (and code-breaking) change to me. Unless you wanted to separate the interpolation from the getter and force people to manually interpolate the values and return something like `NotImplementedError` beforehand.

Reply:
I think I misinterpreted what you said. You meant add it to the `Network.__init__` under kwargs, right? That would work as well, but it seems a bit awkward to me to define it there. It would accomplish the same thing though.